### PR TITLE
Update require-padding-newlines-in-blocks.js fo #1899

### PR DESCRIPTION
**Pull request by cybermerlin**
+ test

### CODE REVIEW

**Comment by hzoo** [comment]
@cybermerlin could you add your tests / this commit to the PR you already made?

**Comment by cybermerlin** [comment]
i did it

**Comment by cybermerlin** [comment]
or I do not know how to do it

File: test/specs/rules/require-padding-newlines-in-blocks.js [hunk 1]
```diff
@@ -470,4 +470,61 @@ describe('rules/require-padding-newlines-in-blocks', function() {
...
+        it('should not report missing extra padding newline after opening brace of function declaration in single line', function() {
+            expect(checker.checkString('function foo() { return bar; }')).to.have.no.errors();
+        });
+
+        it('should not report missing extra padding newline after opening brace of function expression in single line', function() {
+            expect(checker.checkString('var foo = function() { return bar; }')).to.have.no.errors();
+        });
+
+        it('should report missing extra padding newline after opening brace of conditional', function() {
+            expect(checker.checkString('if (true) { abc(); }'))
```
Comment (markelog) [inline test/specs/rules/require-padding-newlines-in-blocks.js]: Why it should omit this? Isn't 
```js
if (true) { doSomething(); }
```
is valid for `singleLine?`

File: test/specs/rules/require-padding-newlines-in-blocks.js [hunk 2]
```diff
@@ -470,4 +470,61 @@ describe('rules/require-padding-newlines-in-blocks', function() {
...
+    describe('open: true, close: false, allExcept: ["singleLine"]', function() {
+        beforeEach(function() {
+            checker.configure({ requirePaddingNewlinesInBlocks: {
+              open: true,
+              close: false,
+              allExcept: ['singleLine'] }
+            });
+        });
+
+        it('should not report missing extra padding newline after opening brace of function declaration in single line', function() {
```
Comment (markelog) [inline test/specs/rules/require-padding-newlines-in-blocks.js]: Could you add one `should report`? It seems one would be needed for the next describe too

File: lib/rules/require-padding-newlines-in-blocks.js [hunk 3]
```diff
@@ -59,7 +60,9 @@
  * if (true) {
  *     doSomething();
  * }
- * if (true) { doSomething(); }
```
Comment (markelog) [inline lib/rules/require-padding-newlines-in-blocks.js]: It seems unnecessary

File: test/specs/rules/require-padding-newlines-in-blocks.js (hunk 1 above)
Comment (cybermerlin) [inline test/specs/rules/require-padding-newlines-in-blocks.js]: yep
if sinle line then not check padding

File: test/specs/rules/require-padding-newlines-in-blocks.js (hunk 2 above)
Comment (cybermerlin) [inline test/specs/rules/require-padding-newlines-in-blocks.js]: yeah

File: lib/rules/require-padding-newlines-in-blocks.js (hunk 3 above)
Comment (cybermerlin) [inline lib/rules/require-padding-newlines-in-blocks.js]: it's right, because it was wrong

Comment (hzoo) [inline lib/rules/require-padding-newlines-in-blocks.js]: If you don't remove it then it's the same code snippet as the one above right?

File: test/specs/rules/require-padding-newlines-in-blocks.js (hunk 1 above)
Comment (markelog) [inline test/specs/rules/require-padding-newlines-in-blocks.js]: So why it's rule should show validation error? If documentation says it shouldn't?

File: test/specs/rules/require-padding-newlines-in-blocks.js (hunk 2 above)
Comment (markelog) [inline test/specs/rules/require-padding-newlines-in-blocks.js]: I mean you have test that proves that error will not be shown, but you don't have one that will

**Comment by markelog** [comment]
Friendly ping

**Comment by markelog** [comment]
Ping us up if want to continue work on this

**Pull request closed without merging**